async.coffee

```diff
@@ -90,11 +90,11 @@ module.exports = async =
 
   detectSeries: Q.promised (arr, iterator, _notFound=undefined) ->
     return Q _notFound if arr.length is 0
-    iterator(arr[0]).then (ok) ->
+    Q(iterator(arr[0])).then (ok) ->
```

Comment:
Make this 
```coffee
Q.try(iterator, arr[0]).then (ok) ->
```
that way if you have an exception it'll get caught properly at toplevel.